services/kvstore/src/main/java/org/hyperledger/besu/services/kvstore/SegmentedInMemoryKeyValueStorage.java

```diff
@@ -270,6 +270,24 @@ public void put(
       removedKeys.computeIfAbsent(segmentIdentifier, __ -> new HashSet<>()).remove(Bytes.wrap(key));
     }
 
+    @Override
+    public Optional<byte[]> get(final SegmentIdentifier segmentIdentifier, final byte[] key) {
+      final Optional<byte[]> updatedValue =
+          updatedValues
+              .computeIfAbsent(segmentIdentifier, __ -> new HashMap<>())
+              .getOrDefault(Bytes.wrap(key), Optional.empty());
+      final Set<Bytes> removedKeys =
+          this.removedKeys.computeIfAbsent(segmentIdentifier, __ -> new HashSet<>());
+
+      // consider the values in order of precedence of removed values, updated values, existing
+      // values
```

Comment:
can the comment explain _why_ this precedence order is needed please? 

It's not clear to me why we need to query removedKeys at all, e.g. what if the same key was added a second time?

Reply:
I was basing this on the order in which we apply the changes in the commit method of `SegmentedKeyValueStorageTransaction` which applies the remove last. So a remove will always override an update being applied last.

The more I think about this I think this get method and the commit method are incorrect and we should be applying the operations in order that they made to the `SegmentedKeyValueStorageTransaction`.